### ISSUE TITLE
Supporting for detection of invalid `select[required]`

### ISSUE THREAD

**Issue by kagankan** [markuplint/markuplint#656]
There are some constraints for `select` element that has `required` attribute.
I hope Markuplint warns if it is invalid.

See:
https://html.spec.whatwg.org/multipage/form-elements.html#attr-select-required

>If a select element has a required attribute specified, does not have a multiple attribute specified, and has a display size of 1; and if the value of the first option element in the select element's list of options (if any) is the empty string, and that option element's parent node is the select element (and not an optgroup element), then that option is the select element's placeholder label option.

# Valid example

```html
<!-- Valid -->
<select required>
    <option value="">Placeholder</option>
    <option value="option1">Option 1</option>
    <option value="option2">Option 2</option>
</select>
```

~~Invalid: has multiple~~

```html
<!-- Valid: if it has `multiple`, it doesn't need to have placeholder label option -->
<select required multiple>
    <option value="">Placeholder</option>
    <option value="option1">Option 1</option>
    <option value="option2">Option 2</option>
</select>
```

~~Invalid: has size other than 1~~

```html
<!-- Valid: if it has `size` with a value greater than 1, it doesn't need to have placeholder label option -->
<select required size="2">
    <option value="">Placeholder</option>
    <option value="option1">Option 1</option>
    <option value="option2">Option 2</option>
</select>
```

# Invalid examples

```html
<!-- Invalid: first option element's value is not empty -->
<select required>
    <option value="placeholder">Placeholder</option>
    <option value="option1">Option 1</option>
    <option value="option2">Option 2</option>
</select>
```

```html
<!-- Invalid: first option element's parent is optgroup -->
<select required>
    <optgroup label="Group">
        <option value="">Placeholder</option>
    </optgroup>
    <option value="option1">Option 1</option>
    <option value="option2">Option 2</option>
</select>
```

**Comment by YusukeHirao** [comment]
It should be created as a rule `placeholder-label-option`. And add `validate` label to it.
